src/tools/cephfs/cephfs-shell

```diff
@@ -949,7 +949,16 @@ class CephFSShell(Cmd):
                     cephfs.unlink(path)
                 except libcephfs.Error as e:
                     # NOTE: perhaps we need a better msg here
-                    set_exit_code_msg(msg=e)
+                    if e.get_error_code() == 2:
+                        set_exit_code_msg(msg="rm: failed to remove "
+                                              f"{path.decode('utf-8')}: "
+                                              "No such file or directory")
+                    elif e.get_error_code() == 21:
+                        set_exit_code_msg(msg="rm: failed to remove "
+                                              f"{path.decode('utf-8')}: "
+                                              "Is a directory")
```

Comment:
You need to pass error code to set_exit_code_msg(), otherwise it will set exit code to 1 for all these cases. :(

Reply:
You'll need to add EISDIR to `exit_codes` dict in cephfs-shell.

Reply:
That would require re-arrangement of the error codes in dict as 21 has been assigned to `errno.ECONNRESET` already.

Reply:
Let's just follow the trend (for now) and add it to the end of that dict?

Reply:
with a random number?

Reply:
That's how it is right now in cephfs-shell IIUC.

Reply:
Okay, tell me the int value to go with, for the key of `errno.EISDIR` :)

Reply:
Done, `errno.ISDIR: 23` added